pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -208,8 +208,11 @@ func (ssc *defaultStatefulSetControl) getStatefulSetRevisions(
 			return nil, nil, err
 		}
 	} else {
+		if set.Status.CollisionCount == nil {
+			set.Status.CollisionCount = new(int64)
```

Comment:
This is in general not safe. You usually want to deep copy any object from the cache prior to mutating it. In this call path you are already working on a copy, so you should be safe, but you should note this in a comment.